test-data/unit/check-unreachable-code.test

```diff
@@ -932,10 +932,86 @@ if False:
 
 if False:
     # Ignore obvious type errors
-    assert_never(expect_str(x))
+    assert_never(expect_str(x))  # E: Statement is unreachable
     reveal_type(x)
 [builtins fixtures/exception.pyi]
 
+[case testUnreachableFlagOkWithDeadStatements2]
+# flags: --warn-unreachable
+from typing import NoReturn
+def assert_never(x: NoReturn) -> NoReturn:
+    assert False
+
+def nonthrowing_assert_never(x: NoReturn) -> None: ...
+
+def expect_str(x: str) -> str: pass
+
+x: int
+if False:
+    assert False
+
+if False:
+    raise Exception()
+
+if False:
+    assert_never(x)
+
+if False:
+    nonthrowing_assert_never(x)  # E: Statement is unreachable
+
+if False:
+    # Ignore obvious type errors
+    assert_never(expect_str(x))
+[builtins fixtures/exception.pyi]
+
+
+[case testSeveralUnreachableStatements]
+# flags: --warn-unreachable
+from typing import NoReturn
+def assert_never() -> NoReturn:
+    assert False
+
+def first() -> NoReturn:
+    assert_never()
+    assert_never()  # E: Statement is unreachable
+    reveal_type(first)
+
+def second() -> NoReturn:
+    raise Exception
+    raise Exception  # E: Statement is unreachable
+    raise Exception()
+    reveal_type(first)
+
+def third() -> NoReturn:
+    assert False
+    assert False  # E: Statement is unreachable
+    assert False
+    assert False
+    reveal_type(first)
+[builtins fixtures/exception.pyi]
+
+[case testSeveralUnreachableStatements2]
+# flags: --warn-unreachable
+from typing import NoReturn
+def assert_never() -> NoReturn:
+    assert False
+
+def first() -> NoReturn:
+    assert_never()
+    assert_never()
+
+def second() -> NoReturn:
+    raise Exception
+    raise Exception
```

Comment:
I'm wondering if this should be treated as unreachable code as well. Maybe the empty statement check for the following statements should be stricter, so that only `pass`, `...` and maybe `reveal_type(...)` would be considered as empty, and maybe asserts.

Reply:
I think that the initial motivation was to allow some "debug" code  to reduce the number of false positives. For example, I can see case where people would use code like:
- `assert False, 'unreachable'`
- `raise RuntimeError('should not be reached')`
- `sys.exit(1)   # should not be reached`